sw/device/tests/sim_dv/spi_tx_rx_test.c

```diff
@@ -0,0 +1,392 @@
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxf,
+                                  kDifPlicMaxPriority) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxlvl,
+                                  kDifPlicMaxPriority) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxlvl,
+                                  kDifPlicMaxPriority) == kDifPlicOk,
+        , "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxerr,
+                                  kDifPlicMaxPriority) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(
+      dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxoverflow,
+                                kDifPlicMaxPriority) == kDifPlicOk,
+      "dif_plic_irq_priority_set failed");
+  CHECK(
+      dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxunderflow,
+                                kDifPlicMaxPriority) == kDifPlicOk,
+      "dif_plic_irq_priority_set failed");
+
+  // Set the threshold for the Ibex to 0.
+  CHECK(dif_plic_target_threshold_set(plic, kTopEarlgreyPlicTargetIbex0, 0x0) ==
+            kDifPlicOk,
+        "dif_plic_target_threshold_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxf,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxlvl,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxlvl,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxerr,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxoverflow,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxunderflow,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+}
+
+static bool exp_irqs_fired() {
+  return (fired_irqs[kDifSpiDeviceIrqTypeRxAboveLevel] &&
```

Comment:
You don't need to include the parenthesis around a return value.

Reply:
Thanks for pointing this out, updated.